lib/ansible/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -150,6 +158,15 @@
       - C(storage) is the storage identifier where to create the disk.
       - C(size) is the size of the disk in GB.
       - C(format) is the drive's backing file's data format. C(qcow2|raw|subvol).
+  ipconfig:
```

Comment:
If this option takes a dictionary of options, you can define them using suboptions. See [iptables.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/system/iptables.py#L541) for an example of how to use suboptions.

Reply:
I don't think this is possible to do with sub-options, there are more options in this module like ipconfig (net, scsi, serial for example), they all share the same characteristics. I've just copied what the module already had for these types of parameters.

Reply:
This module may have been created before we had sub-option support in Ansible. While there are a number of options in this module that do accept a dictionary without any type of validation, it would be nice if the new `ipconfig` used sub-options as a model for how the other options should be implemented.

The module UI would look the same, but you would get built-in validation. The data essentially exists in the docs, you would just be enforcing it with code as well.

```python
ipconfig=dict(type='dict',
              options=dict(
                   gw=dict(type='string'),
                   gw6=dict(type='string'),
                   ip=dict(type='string'),
                   ip6=dict(type='string')
                ),
```

Feel free to correct me if I'm not fully understanding why this won't work.

Reply:
ipconfig is structured like this:
```
{
 'ipconfig': {
   'ipconfig0': {'gw':'192.168.1.1', 'gw6': None, 'ip': '192.168.1.2', 'ip6': None}
  }
}
```
I don't know if the same is possible with suboptions, since ipconfig0 is specifically named key. With your solution (if I'm understanding correctly). The data format will be:

```
{
 'ipconfig': {
   {'gw':'192.168.1.1', 'gw6': None, 'ip': '192.168.1.2', 'ip6': None}
  }
}
```
Which is incorrect.

Correct me if I'm wrong please.